libvirt/tests/src/virsh_cmd/domain/virsh_cpu_stats.py

```diff
@@ -122,18 +136,30 @@ def run(test, params, env):
             if get_noopt or get_total:
                 mt_end = re.search('Total', output).end()
                 total_list = output[mt_end + 1:].split()
-
-                total_time = int(total_list[1])
-                user_time = int(total_list[4])
-                system_time = int(total_list[7])
-
-                # check Total cpu_time >= User + System cpu_time
-                if user_time + system_time > total_time:
-                    test.fail("total cpu_time < user_time + "
-                              "system_time")
-                logging.debug("Check total cpu_time %d >= user + system "
-                              "cpu_time %d",
-                              total_time, user_time + system_time)
+                total_time = float(total_list[1])
+                user_time = float(total_list[4])
+                system_time = float(total_list[7])
+
+                # check libvirt cpu_time and system_time between the pre and the next cgroup time
+                # unit conversion (Unit: second)
+                pre_user_time = int(cpuacct_res_pre[1])/100
+                pre_sys_time = int(cpuacct_res_pre[3])/100
+                next_user_time = int(cpuacct_res_next[1])/100
+                next_sys_time = int(cpuacct_res_next[3])/100
+
+                # check user_time
+                if next_user_time >= user_time >= pre_user_time:
+                    logging.debug("The libvirt user_time:%s should between cgroup pre user_time:%s and cgroup next user_time:%s ",
+                                  user_time, pre_user_time, next_user_time)
```

Comment:
I would suggest printing a simple message if the check passes and providing details if it fails.

```suggestion
                    logging.debug("Got the expected user_time: %s", user_time)
```